File: frontend/src/services/rentService.js
import api from "./api";

const getPropertyRents = async (propertyId) => {
    const response = await api.get(`rent/property/${propertyId}`);
    return response.data;
};

const getRentSummary = async (propertyId) => {
    const response = await api.get(`rent/property/${propertyId}/summary`);
    return response.data;
};

const getRentDetails = async (rentId) => {
    const response = await api.get(`rent/${rentId}`);
    return response.data;
};

const recordPayment = async (rentId, paymentRequest) => {
    const response = await api.post(`rent/${rentId}/pay`,paymentRequest);
    return response.data;
};

const getMyRent = async() => {
    const response = await api.get("rent/my-rent");
    return response.data;
};


const rentService= {
    getPropertyRents,
    getRentSummary,
    getRentDetails,
    recordPayment,
    getMyRent
};

export default rentService;

File: backend/src/main/java/com/nishtha/RoomEase/rent/service/RentService.java
package com.nishtha.RoomEase.rent.service;

import com.nishtha.RoomEase.rent.dto.RecordPaymentRequest;
import com.nishtha.RoomEase.rent.dto.RentDetailsResponse;
import com.nishtha.RoomEase.rent.dto.RentResponse;
import com.nishtha.RoomEase.rent.dto.RentSummaryResponse;

import java.util.List;

public interface RentService {

    List<RentResponse> getPropertyRents(Long propertyId);

    RentSummaryResponse getRentSummary(Long propertyId);

    RentDetailsResponse getRentDetails(Long rentId);

    RentResponse recordPayment(Long rentId, RecordPaymentRequest request);

    void generateMonthlyRent();

    void updateOverdueRents();

    List<RentDetailsResponse> getMyRent();

}